gui/src/daemon/embedded.rs

```diff
@@ -40,71 +30,36 @@ impl Daemon for EmbeddedDaemon {
         false
     }
 
-    fn load_config(&mut self, cfg: Config) -> Result<(), DaemonError> {
-        if self.handle.is_none() {
-            return Ok(());
-        }
-
-        let next = DaemonHandle::start_default(cfg).map_err(DaemonError::Start)?;
-        self.handle.take().unwrap().into_inner().unwrap().shutdown();
-        self.handle = Some(RwLock::new(next));
-        Ok(())
-    }
-
     fn config(&self) -> Option<&Config> {
         Some(&self.config)
     }
 
-    fn stop(&mut self) -> Result<(), DaemonError> {
-        if let Some(h) = self.handle.take() {
-            let handle = h.into_inner().unwrap();
-            handle.shutdown();
+    fn stop(&self) {
+        self.handle.trigger_shutdown();
+        while !self.handle.shutdown_complete() {
+            tracing::debug!("Waiting daemon to shutdown");
+            std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
nit, let's avoid triggering the "process stuck"
```suggestion
            tracing::debug!("Waiting for daemon to shutdown");
            std::thread::sleep(std::time::Duration::from_millis(500));
```